addons/website_gengo/controllers/main.py

```diff
@@ -37,3 +37,30 @@ def set_gengo_config(self,config):
     def post_gengo_jobs(self):
         request.registry['base.gengo.translations']._sync_request(request.cr, request.uid, limit=GENGO_DEFAULT_LIMIT, context=request.context)
         return True
+
+    @http.route('/website/set_translations', type='json', auth='user', website=True)
+    def set_translations(self, data, lang):
+        IrTranslation = request.env['ir.translation']
+        for term in data:
+            initial_content = term['initial_content'].strip()
+            tid = term['translation_id']
+            if not tid:
+                translations = IrTranslation.search_read([('lang', '=', lang), ('src', '=', initial_content)], fields=['id'])
+                if translations:
+                    tid = map(lambda t_id: t_id['id'], translations)
```

Comment:
You may get more than one record for same src, what if you get more then one record for same source, should all those entries updated(if yes then shouldn't it be translation_ids) ?